middleware/gateway/unittest/source/test_manager.cpp

```diff
@@ -1271,6 +1271,85 @@ namespace casual
 
       }
 
+      TEST( gateway_manager_outbound, transaction_resource_error_reply_on_connection_lost_before_commit_request)
+      {
+         common::unittest::Trace trace;
+
+         auto b = local::domain( R"(
+domain: 
+   name: B
+   groups:
+      -  name: user
+         dependencies: [ base]
+   
+   servers:
+      - path: ${CASUAL_MAKE_SOURCE_ROOT}/middleware/example/server/bin/casual-example-server
+        memberships: [ user]
+
+   gateway:
+      inbound:
+         groups:
+            -  connections: 
+                  -  address: 127.0.0.1:7012
+)");
+
+         // get a handle for b:s inbound
+         auto inbound = unittest::fetch::until( unittest::fetch::predicate::listeners( 1)).inbound.groups.at( 0).process;
+         ASSERT_TRUE( inbound);
+
+         auto a = local::domain( R"(
+domain: 
+   name: A
+   gateway:
+      outbound:
+         groups:
+            -  connections:
+                  -  address: 127.0.0.1:7012
+                     services: [ casual/example/domain/echo/B]
+)");
+
+         unittest::fetch::until( unittest::fetch::predicate::outbound::connected( 1));
+         
+         // call service in B in transaction
+         {
+            using namespace std::literals;
+
+            EXPECT_EQ( transaction::context().begin(), code::tx::ok);
+
+            buffer::Payload payload;
+            payload.type = "X_OCTET/";
+            common::algorithm::copy( "casual"sv, std::back_inserter( payload.memory));
+
+            auto result = common::service::call::context().sync( "casual/example/domain/echo/B", common::buffer::payload::Send{ payload}, {});
+
+            EXPECT_TRUE( result.buffer.memory == payload.memory);
+
+         }
+
+         // we 'lose' our connection to b
+         {
+            common::signal::send( inbound.pid, code::signal::terminate);
+         }
+
+         // we try to commit our transaction
+         {
```

Comment:
There is a timing issue here. There is no guarantee when signals are delivered. You should check that the _outbound_ is non-existent, before committing.
Hence, keep a process, as your inbound, and then _fetch::until_ you can't find it in state